docs/implementation.md

```diff
@@ -0,0 +1,19 @@
+# Implementation
+
+```{eval-rst}
+.. currentmodule:: scipy_doctest.impl
+.. autosummary::
+    :nosignatures:
+    :toctree: generated
+
+    DTConfig
+    DTChecker
+    DTRunner
+    DebugDTRunner
+    DTFinder
+    DTParser
+    try_convert_namedtuple
+    try_convert_printed_array
+    has_masked
+    try_split_shape_from_abbrv
```

Comment:
`DT*` classes are definitely for the user, the rest is implementation details, I'd say.